internal/provider/providers/porkbun/provider.go

```diff
@@ -119,46 +119,76 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 		recordType = constants.AAAA
 	}
 	ipStr := ip.String()
-	recordIDs, err := p.getRecordIDs(ctx, client, recordType)
+	records, err := p.getRecords(ctx, client, recordType, p.owner)
 	if err != nil {
 		return netip.Addr{}, fmt.Errorf("getting record IDs: %w", err)
 	}
 
-	if len(recordIDs) == 0 {
-		// ALIAS record needs to be deleted to allow creating an A record.
-		err = p.deleteALIASRecordIfNeeded(ctx, client)
-		if err != nil {
-			return netip.Addr{}, fmt.Errorf("deleting ALIAS record if needed: %w", err)
-		}
-
-		err = p.createRecord(ctx, client, recordType, ipStr)
-		if err != nil {
-			return netip.Addr{}, fmt.Errorf("creating record: %w", err)
+	if len(records) > 0 {
+		for _, record := range records {
+			err = p.updateRecord(ctx, client, recordType, p.owner, ipStr, record.ID)
+			if err != nil {
+				return netip.Addr{}, fmt.Errorf("updating record: %w", err)
+			}
 		}
 		return ip, nil
 	}
 
-	for _, recordID := range recordIDs {
-		err = p.updateRecord(ctx, client, recordType, ipStr, recordID)
+	// For new domains, Porkbun Creates 2 Default DNS Records which point to their "parked" domain page:
+	// ALIAS domain.tld -> pixie.porkbun.com
+	// CNAME *.domain.tld -> pixie.porkbun.com
+	// ALIAS and CNAME records conflict with A and AAAA records, and attempting to create an A or AAAA record
+	// will return a 400 error if they aren't first deleted.
+	porkbunParkedDomain := "pixie.porkbun.com"
+	switch {
+	case p.owner == "@":
+		// Delete ALIAS domain.tld -> pixie.porkbun.com record
+		err = p.deleteSingleMatchingRecord(ctx, client, constants.ALIAS, "@", porkbunParkedDomain)
+		if err != nil {
+			return netip.Addr{}, fmt.Errorf("deleting default parked domain record: %w", err)
+		}
+	case p.owner == "*":
+		// Delete ALIAS domain.tld -> pixie.porkbun.com record
+		// Error is ignored as the ALIAS could be set to something besides the parked domain. Failure here is non-fatal.
+		_ = p.deleteSingleMatchingRecord(ctx, client, constants.ALIAS, "@", porkbunParkedDomain)
```

Comment:
Eh it should be a fatal error if we get an API error for example, so I changed it to

```go
		err = p.deleteSingleMatchingRecord(ctx, client, constants.ALIAS, "@", porkbunParkedDomain)
		if err != nil && !stderrors.Is(err, errors.ErrConflictingRecord) {
			// allow conflict ALIAS records to be set to something besides the parked domain
			return fmt.Errorf("deleting default ALIAS @ parked domain record: %w", err)
		}
```